apollo-api/api/apollo-api.api

```diff
@@ -29,7 +29,7 @@ public final class com/apollographql/apollo3/api/Adapters {
 }
 
 public final class com/apollographql/apollo3/api/ApolloRequest : com/apollographql/apollo3/api/HasExecutionContext {
-	public fun <init> (Lcom/apollographql/apollo3/api/Operation;Ljava/util/UUID;Lcom/apollographql/apollo3/api/ExecutionContext;)V
+	public synthetic fun <init> (Lcom/apollographql/apollo3/api/Operation;Ljava/util/UUID;Lcom/apollographql/apollo3/api/ExecutionContext;Lkotlin/jvm/internal/DefaultConstructorMarker;)V
```

Comment:
I can understand the `synthetic` (might be the way kotlin hides private constructors?) but the `DefaultConstructorMarker` is weird here. There's no default parameter at all 🤔

Reply:
Looking at the rest of the file, this alternative `<init>` with `synthetic` + `DefaultConstructorMarker` parameter  is there "often but not all the time" 🤔. I wonder what the rule (and use) are.

Reply:
Turns out the `DefaultConstructor` marker is not (always) related to default parameters. It's just a marker (hence the name) to avoid any name clash with other possibly defined constructor.

Still weirds that it ends up in the public API. I opened an issue there: https://github.com/Kotlin/binary-compatibility-validator/issues/73

Reply:
Indeed the constructor is `private` so it would be expected to not show up